core/src/main/java/org/apache/brooklyn/core/BrooklynFeatureEnablement.java

```diff
@@ -61,6 +61,36 @@
     /** whether feeds are automatically registered when set on entities, so that they are persisted */
     public static final String FEATURE_FEED_REGISTRATION_PROPERTY = FEATURE_PROPERTY_PREFIX+".feedRegistration";
 
+    /**
+     * <p>
+     * Enabling CORS is strongly discouraged.
```

Comment:
CORS per se is a good thing. When combined with some other features like basic auth could be a terrible idea because the credentials need to be shared with all clients. Much easier for malicious users to extract the credentials from the javascript env and use them any way they like.
Basic auth would be the reason why I'd stick to proxy until we get something smarter like web tokens. 

I like @geomacy's suggestion above, but perhaps with the proxy recommendation added to it.